lib/retry.js

```diff
@@ -4,37 +4,67 @@ const sleep = msec => new Promise(resolve => {
   setTimeout(resolve, msec);
 });
 
-const promisify = fn => (...args) => (
-  new Promise((resolve, reject) => (
-    fn(...args, (err, data) => (
-      err ? reject(err) : resolve(data)
-    ))
-  ))
-);
+const promisify = (fn, args) => new Promise((resolve, reject) => {
```

Comment:
Maybe make a separate file 'utils' and save promisify, deepClone etc. there?

Reply:
Fixed, thank you.